File: ed1/finale/dadi/att/dadi.c
#include <stdio.h>
#include <assert.h>

#define MAXN 100000

void simula(int N, char C[], int R[]) {
    // Mettete qui il codice della soluzione
}

char C[MAXN];
int R[MAXN];

int main() {
    FILE *fr, *fw;
    int N, M, i;

    fr = fopen("input.txt", "r");
    fw = fopen("output.txt", "w");
    assert(1 == fscanf(fr, "%d\n", &N));
    M = 0;
    for (i = 0; i < N; ++i) {
        assert(1 == fscanf(fr, "%c", &C[i]));
        if (C[i] == 'T' || C[i] == 'F' || C[i] == 'R')
            ++M;
    }

    simula(N, C, R);
    for (i = 0; i < M; ++i)
        fprintf(fw, "%d ", R[i]);
    fprintf(fw, "\n");

    fclose(fr);
    fclose(fw);
    return 0;
}

File: ed1/finale/dadi/att/dadi.pas
procedure simula(N: longint; var C: array of char; var R: array of longint);
begin
    (* Mettete qui il codice della soluzione *)
end;


const
    MAXN = 100000;
var
    N, i, j : longint;
    C       : array[0..MAXN] of char;
    R       : array[0..MAXN-1] of longint;
    fr, fw  : text;

begin
    assign(fr, 'input.txt');
    assign(fw, 'output.txt');
    reset(fr);
    rewrite(fw);
    readln(fr, N);
    readln(fr, C);

    simula(N, C, R);
    j := 0;
    for i:=0 to N-1 do if (C[i] = 'T') or (C[i] = 'F') or (C[i] = 'R') then begin
        write(fw, R[j], ' ');
        inc(j);
    end;
    writeln(fw);
    close(fr);
    close(fw);
end.
